src/store/getters.ts

```diff
@@ -32,13 +32,24 @@ const getters: GetterTree<VQBState, any> = {
   /**
    * a direct "usable" index (i.e. convert "-1" to a positive one) of last active step.
    */
-  computedActiveStepIndex: (state: VQBState) =>
-    state.selectedStepIndex === -1 ? currentPipeline(state).length - 1 : state.selectedStepIndex,
+  computedActiveStepIndex(state: VQBState) {
+    const pipeline = currentPipeline(state);
+    if (!pipeline) {
+      return;
+    }
+    return state.selectedStepIndex === -1 ? pipeline.length - 1 : state.selectedStepIndex;
+  },
   /**
    * the first step of the pipeline. Since it's handled differently in the UI,
    * it's useful to be able to access it directly.
    */
-  domainStep: (state: VQBState) => currentPipeline(state)[0],
+  domainStep(state: VQBState) {
+    const pipeline = currentPipeline(state);
```

Comment:
I think you could use the conditional operator here, as in:

```typescript
return currentPipeline(state)?[0];
```

Reply:
Oh, I didn't know that landed in Ts ! Nice to know, this will greatly simplify this PR :D